core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -271,4 +337,69 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
         }
     )
 
+  // approximate the byte size of this entity by looking at the character length of its JSONized attributes
+  private def calculateEntitySize(entity: Entity): Int =
+    CompactEntitySerialization.toSql(entity.attributes).compactPrint.length
+
+  private def insertBatch(batch: Seq[Entity]): ReadWriteAction[Seq[Entity]] =
+    for {
+      // batch insert to ENTITY table. Save the whole batch first to handle cases where an entity in this batch
+      // has a reference to another entity in the same batch.
```

Comment:
what about references to entities in a future batch?

Reply:
Inserting records out of order will fail with a reference-not-found error, same as the current implementation. I think it would be a lot easier to make this case succeed if we wanted to, but am not attempting to do that yet.